### PR TITLE
tests/end-to-end-test-for-report-submission-using-playwright-in-ci

### PR DESCRIPTION
**Pull request by wssheldon**
Tests work great locally, need to brainstorm how to fully automate authentication in CI, and how to run the development server in CI on GH Actions.

I recreated the bug where the "Load More" button did not appear for the Project select button, introduced [here](https://github.com/Netflix/dispatch/pull/2770). The test will check for the "Load More" text and fail if it can't be found after selecting the `Type` button, this failed as expected:

```typescript
    // check that the 'Load More' selector is available upon opening the Type dropdown
    await expect(
      page.getByText("Load More"),
      "The 'Load More' selector should be visibile."
    ).toBeVisible()
```

```bash
Error: The 'Load More' selector should be visibile.

Call log:
  - expect.toBeVisible with timeout 10000ms
  - waiting for getByText('Load More')
  -   unexpected value "hidden"

  35 |       page.getByText("Load More"),
  36 |       "The 'Load More' selector should be visibile."
> 37 |     ).toBeVisible()
     |       ^
  38 |
  39 |     await page.getByText(type, { exact: true }).click()

    at /Users/wshel/Projects/dispatch/tests/e2e/report-submission.spec.ts:37:7
```

### CODE REVIEW

**Comment by wssheldon** [comment]
> I think you should be able to run these via a github action:

I think I'll need to use the `.devcontainer` to be able to run `dispatch server develop`

File: tests/e2e/report-submission.spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    await page.getByRole("combobox").filter({ hasText: "Project" }).locator("i").click()
+    await page.getByText(project, { exact: true }).click()
+
+    // Type
+    await page.getByRole("button", { name: "Type" }).click()
+    await page.getByText(type, { exact: true }).click()
+
+    // Priority
+    await page.getByRole("button", { name: "Priority" }).click()
+    await page.getByText(priority).click()
```
Comment (mvilanova) [inline tests/e2e/report-submission.spec.ts:36]: Shouldn't these be of type combobox or are you trying to mock clicking on it and loading a value?

Comment (wssheldon) [reply tests/e2e/report-submission.spec.ts:36]: This clicks the button and types a value.

Comment (wssheldon) [reply tests/e2e/report-submission.spec.ts:36]: It could be done either way though like you said.

File: tests/e2e/report-submission.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    // Open the dropdown selection for the Tags field
+    await page.getByRole("combobox").filter({ hasText: "Tags" }).locator("i").click()
+
+    // Soft check that the 'Load More' selector is available upon opening the Tags dropdown
+    await expect
+      .soft(page.getByText("Load More"), "The 'Load More' selector should be visibile.")
```
Comment (wssheldon) [inline tests/e2e/report-submission.spec.ts:43]:
```suggestion
      .soft(page.getByText("Load More"), "The 'Load More' selector should be visible when there are more than 5 options in the Tags combobox.")
```

Also, can we check if there's more than 5 options in a smart way first to make this a conditional check?

File: tests/e2e/report-submission.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    // Wait for the incident to be created
+    await page.waitForLoadState("networkidle")
+
+    // Soft validate that we get redirected to the incident submission form
+    await expect
+      .soft(page)
+      .toHaveURL(
+        encodeURI(
+          `./default/incidents/report?project=${project}&incident_priority=${priority}&incident_type=${type}`
+        )
```
Comment (wssheldon) [inline tests/e2e/report-submission.spec.ts:61]: The [custom exception messages](https://github.com/Netflix/dispatch/pull/2936/files#diff-5221afac34a469ab3bdfa7bd61adfa4d2093bb89bd78506d6dfd48434a917280R43) are pretty nice. We should consider adding one here.

File: tests/e2e/report-submission.spec.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    // Soft validate that we recieve the report form.
+    await expect.soft(page.getByText("Incident Report")).toBeVisible()
+
+    // Soft validate that we recieve the report form.
+    await expect
+      .soft(
+        page.getByText(
+          "This page will be populated with incident resources as they are created (if available). If you have any questions, please feel free to review the Frequently Asked Questions (FAQ) document linked below, and/or reach out to the listed Incident Commander."
+        )
```
Comment (wssheldon) [inline tests/e2e/report-submission.spec.ts:72]: The [custom exception messages](https://github.com/Netflix/dispatch/pull/2936/files#diff-5221afac34a469ab3bdfa7bd61adfa4d2093bb89bd78506d6dfd48434a917280R43) are pretty nice. We should consider adding one here.

File: tests/e2e/report-submission.spec.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  })
+  test("Should allow me to report an incident", async ({ page }) => {
+    /* The ability to report an incident is one of the most critical
+      user stories in the Dispatch application. */
+
+    const project = "Test"
+    const type = "Customer Data"
+    const priority = "Low"
+
+    await page.getByRole("link", { name: "Report Incident" }).click()
```
Comment (wssheldon) [inline tests/e2e/report-submission.spec.ts:16]: I wonder if there might be a reason to visit the URL directly as opposed to clicking the `Report Incident` button. I suspect the direct URL visit may be more reliable and this test might be better isolated if we avoid any side effects of this button click.

File: tests/e2e/report-submission.spec.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    // Soft validate that we recieve the report form.
+    await expect
+      .soft(
+        page.getByText(
+          "This page will be populated with incident resources as they are created (if available). If you have any questions, please feel free to review the Frequently Asked Questions (FAQ) document linked below, and/or reach out to the listed Incident Commander."
+        )
+      )
+      .toBeVisible()
+  })
```
Comment (wssheldon) [inline tests/e2e/report-submission.spec.ts:75]: We might want to extend this test to validate the submission report card has not regressed in anyway. This pattern will probably come up a lot and it feels very verbose and maybe flakey to do it this way. Maybe a todo could be to investigate using something like this: https://www.thoughtworks.com/en-us/radar/techniques/component-visual-regression-testing

Comment (wssheldon) [reply tests/e2e/report-submission.spec.ts:75]: https://storybook.js.org/tutorials/intro-to-storybook/vue/en/get-started/

File: src/dispatch/static/dispatch/package-lock.json [hunk 7]
```diff
@@ -7047,8 +7085,7 @@
       "version": "18.11.7",
       "resolved": "https://registry.npmjs.org/@types/node/-/node-18.11.7.tgz",
       "integrity": "sha512-LhFTglglr63mNXUSRYD8A+ZAIu5sFqNJ4Y2fPuY7UlrySJH87rRRlhtVmMHplmfk5WkoJGmDjE9oiTfyX94CpQ==",
-      "dev": true,
-      "peer": true
```
Comment (wssheldon) [inline src/dispatch/static/dispatch/package-lock.json:7051]: Not sure why that changed

**Pull request merged**